client/src/sessions/password.rs

```diff
@@ -1,14 +1,43 @@
 use crate::sessions::Session;
-use tpm2_rs_base::errors::{TssResult, TssTcsError};
+use tpm2_rs_base::errors::{TpmRcResult, TssResult, TssTcsError};
 use tpm2_rs_base::{
     Tpm2bAuth, Tpm2bNonce, Tpm2bSimple, TpmaSession, TpmiShAuthSession, TpmsAuthCommand,
     TpmsAuthResponse,
 };
 
 /// A password session.
+///
+/// # Usage:
+/// ```
+/// use tpm2_rs_client::sessions::PasswordSession;
+///
+/// let password1 = PasswordSession::new("hello world").unwrap(); // from a string
+/// let password2 = PasswordSession::new([1, 2, 3, 4, 5, 6]).unwrap(); // from a byte array
+///
+/// assert_eq!(password1.get_secret(), b"hello world");
+/// assert_eq!(password2.get_secret(), &[1, 2, 3, 4, 5, 6]);
+/// ```
 #[derive(Debug, PartialEq, Default)]
 pub struct PasswordSession {
-    pub auth: Tpm2bAuth,
+    auth: Tpm2bAuth,
+}
+
+impl PasswordSession {
+    /// This function creates a new password session using the
+    /// specified password.
+    ///
+    /// # Errors:
+    /// Returns [`tpm2_rs_base::errors::TpmRcError::Size`] if the
+    /// password size in bytes exceeds [`Tpm2bAuth::MAX_BUFFER_SIZE`].
+    pub fn new<T: AsRef<[u8]>>(password: T) -> TpmRcResult<Self> {
```

Comment:
We don't *need* to take this by value, but this technically would allow for code like `PasswordSession::new(String::from("wow"))` (or other similar by-value constructs). I'm not sure if we want this or if we want to enforce that you are passing in a reference to some value (since we're copying the data anyways).

If you want to enforce references, simply: `(password: &T)`, and then that will disallow code like `PasswordSession::new(String::from("foo"))` (at least without the caller explicitly saying to pass it by-reference like this: `&String::from("foo")`).